### PR TITLE
change structure and start issue 3

### PR DESCRIPTION
**Pull request by DaveTheWebDev**
Structure changed, added some folders, changed CSS to SCSS, started issue 3#, added function renderAndAppendElement

### CODE REVIEW

**Review by ZibiH — COMMENTED** [review]
Everything looks OK, only missing 'i' in scss file name '_variables'

**Review by ZibiH — COMMENTED** [review]
I like the changes, I think we can and should merge it ASAP

File: src/app/App.js [hunk 1]
```diff
@@ -1,4 +1,8 @@
+import game from './Game'
+
+
 export const App = ({options}) => {
+  game.start()
```
Comment (pspaczek) [inline src/app/App.js:5]: Question (Q): do you want to do something more in this function? I think here you should build a complete view here and keep the logic in deeper components :)

File: src/app/Game.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import menuOptionsObj from './components/MenuOptions';
```
Comment (pspaczek) [inline src/app/Game.js:1]: IMO you can do this in App.js, it will be the initial view of your app :)

File: src/app/Game.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+class Game {
+  constructor() {}
+  start() {
+    const menuOptions = menuOptionsObj.render()
```
Comment (pspaczek) [inline src/app/Game.js:6]: R: maybe you can pass listener ass menu options like new menuOptionsObj({class: 'button button--big', on:()=>{...}}) and keep fully responsibility inside these component. Now you have a few place when you have logic for this component :)

File: src/app/components/MenuOptions.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import renderElement from '../tools/renderElement'
+
+class MenuOptions {
+  constructor() {
+    this.options = ['People', 'Vehicles', 'Starships']
+  }
+  render() {
```
Comment (pspaczek) [inline src/app/components/MenuOptions.js:7]: R: add event listeners here from component config passed in App :)

File: src/app/components/MenuOptions.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+      const allOptions = document.querySelectorAll('.options__mode');
+      allOptions.forEach(option => option.classList.remove('options__mode--active'))
+      const activeMode = e.target
+      activeMode.classList.add('options__mode--active')
+      callback(activeMode.dataset.mode)
+    }
+  }
+
+  changeMode(activeMode) {
+    if (activeMode === 'people') {
```
Comment (pspaczek) [inline src/app/components/MenuOptions.js:31]: do you need if/else? you return activeMode -> activeMode === 'people' return 'people' no idea why we need if. But, if you want to do something more then use switch ;)

File: src/app/tools/renderElement.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...const possibleHTMLTags = ["a", "abbr", "acronym", "address", "applet", "area", "article", "aside", "audio", "b", "base", "basefont", "bdi", "bdo", "bgsound", "big", "blink", "blockquote", "body", "br", "button", "canvas", "caption", "center", "cite", "code", "col", "colgroup", "content", "data", "datalist", "dd", "decorator", "del", "details", "dfn", "dir", "div", "dl", "dt", "element", "em", "embed", "fieldset", "figcaption", "figure", "font", "footer", "form", "frame", "frameset", "h1", "h2", "h3", "h4", "h5", "h6", "head", "header", "hgroup", "hr", "html", "i", "iframe", "img", "input", "ins", "isindex", "kbd", "keygen", "label", "legend", "li", "link", "listing", "main", "map", "mark", "marquee", "menu", "menuitem", "meta", "meter", "nav", "nobr", "noframes", "noscript", "object", "ol", "optgroup", "option", "output", "p", "param", "plaintext", "pre", "progress", "q", "rp", "rt", "ruby", "s", "samp", "script", "section", "select", "shadow", "small", "source", "spacer", "span", "strike", "strong", "style", "sub", "summary", "sup", "table", "tbody", "td", "template", "textarea", "tfoot", "th", "thead", "time", "title", "tr", "track", "tt", "u", "ul", "var", "video", "wbr", "xmp"]
```
Comment (pspaczek) [inline src/app/tools/renderElement.js:1]: looks nice, maybe it can add listeners too :)

File: styles/components/_MenuOptions.scss [hunk 7]
```diff
@@ -0,0 +1,25 @@
+.options {
+  display: flex;
+  justify-content: space-around;
+  align-items: center;
+  width: 1200px;
```
Comment (pspaczek) [inline styles/components/_MenuOptions.scss:5]: use rem instead of px

File: styles/components/_MenuOptions.scss [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  height: 180px;
+  border-radius: 16px;
+  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25), 4px 4px 40px rgba(255, 0, 0, 0.9);
+  &__mode {
+    padding-bottom: 10px;
+    font-size: 42px;
+    font-weight: 600;
+    cursor: pointer;
+    &--active::after {
```
Comment (pspaczek) [inline styles/components/_MenuOptions.scss:14]: for me 👇 this looks cleaner 
```
&--active {
  &::after {
    ...
  }
}
```

File: src/app/tools/renderElement.js [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
+const quizMainNode = document.getElementById('swquiz-app');
+
+const renderElement = (tagHTML, className, innerText = '', parentElement = quizMainNode, ) => {
```
Comment (pspaczek) [inline src/app/tools/renderElement.js:5]: WDYT about destructuring and use object to pass options :)

**Review by pspaczek — CHANGES_REQUESTED** [review]

**Pull request merged**